### ISSUE TITLE
amb (race) as a new FlattenStrategy

### ISSUE THREAD

**Issue by inamiy** [ReactiveCocoa/ReactiveSwift#138]
Just got inspired from https://github.com/ReactiveCocoa/ReactiveObjC/pull/51, I think we can add `FlattenStrategy.first` where multiple producers race and winner forwards all the events after they get `flatten`ed.

Diagram: http://reactivex.io/documentation/operators/amb.html

I think this can be useful in some scenario (though I have never used it before).

**Comment by NachoSoto** [comment]
Wouldn't that be equivalent to
```swift
nestedSignal
  .take(1)
  .flatten(.Merge) // or anything else
```

**Comment by mdiep** [comment]
No, I think it's the first flattened signal to send an event.

```swift
let (s1, o1) = Signal.pipe()
let (s2 o2) = Signal.pipe()
let (s3, o3) = Signal.pipe()

Signal(values: s1, s2, s3)
    .flatten(.first)
   .observeValues { print($0) }

o3.send(value: 7) // s3 becomes the chosen signal and its values are used
```

**Comment by inamiy** [comment]
One example would be `func timeout` that can be re-implemented like this:

```swift
public func timeout(after interval: TimeInterval, raising error: Error, on scheduler: DateSchedulerProtocol) -> SignalProducer<Value, Error> {
    return SignalProducer(values: [
        self, 
        SignalProducer(error: error).delayError(interval, on: scheduler)
    ]).flatten(.first)
}
// NOTE: `delayError` doesn't exist yet
```

**Comment by RuiAAPeres** [comment]
@inamiy any progress on this? 😊

**Comment by inamiy** [comment]
@RuiAAPeres Not yet, but I will tackle on this soon :)

**Comment by andersio** [comment]
Implemented in #233.
